Alias to array of references doesn't generate pop function correctly.

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, "Manjaro Linux"
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
CC version: cc (GCC) 12.1.0

getwd: /home/adamoates/Documents/GitHub/Cyberian_Tiger
vmodules: /home/adamoates/.vmodules
vroot: /home/adamoates/v
vexe: /home/adamoates/v/v
vexe mtime: 2022-06-26 17:13:28
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 223b96a.5a79a54

Git version: git version 2.36.1
Git vroot status: weekly.2022.24-94-g5a79a54f-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 eee03626

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg ../../test/test.v`
```v
[heap]
struct Attribute {
mut:
	name string
	value string
}

type AttributeStack = []&Attribute

fn main() {
	mut stack := AttributeStack([]&Attribute{})
	stack << &Attribute{'foo', 'bar'}
	dump(stack.pop())
}

```

**What did you expect to see?**

Expected the last item in array to be dumped and popped from array.


**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 000004f5
tcc: error: file 'crt1.o' not found
/tmp/v_1000/test.619390649417170183.tmp.c:12133: error: cannot convert 'struct array' to 'struct array *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1165,8 +1165,7 @@ pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 			c.error('cannot $method_name `$arg_sym.name` to `$left_sym.name`', arg_expr.pos())
 		}
 	} else if final_left_sym.info is ast.Array && method_name in ['first', 'last', 'pop'] {
-		node.return_type = final_left_sym.info.elem_type
-		return node.return_type
+		return c.array_builtin_method_call(mut node, left_type, final_left_sym)
 	} else if c.pref.backend.is_js() && left_sym.name.starts_with('Promise<')
 		&& method_name == 'wait' {
 		info := left_sym.info as ast.Struct

--- a/vlib/v/tests/array_of_alias_pop_test.v
+++ b/vlib/v/tests/array_of_alias_pop_test.v
@@ -1,0 +1,17 @@
+[heap]
+struct Attribute {
+mut:
+	name  string
+	value string
+}
+
+type AttributeStack = []&Attribute
+
+fn test_array_of_alias_pop() {
+	mut stack := AttributeStack([]&Attribute{})
+	stack << &Attribute{'foo', 'bar'}
+	ret := stack.pop()
+	println(ret)
+	assert ret.name == 'foo'
+	assert ret.value == 'bar'
+}

